pycontrails/models/ps_model/ps_model.py

```diff
@@ -0,0 +1,936 @@
+    ----------
+    rn: npt.NDArray[np.float_]
+        Reynolds number
+
+    Returns
+    -------
+    npt.NDArray[np.float_]
+        Skin friction coefficient.
+
+    Notes
+    -----
+    The skin friction coefficient, a dimensionless quantity, is used to estimate the skin friction drag, which is the
+    resistance force exerted on an object moving in a fluid. Given that aircraft at cruise generally experience a
+    narrow range of Reynolds number of between 3E7 and 3E8, it is approximated using a simple power law.
+
+    References
+    ----------
+    - Refer to Eq. (28) of Poll & Schumann (2021).
+    - Poll & Schumann (2021). An estimation method for the fuel burn and other performance characteristics of civil
+        transport aircraft in the cruise. Part 1: fundamental quantities and governing relations for a general
+        atmosphere. Aero. J., 125(1284), 296-340, doi: 10.1017/aer.2020.62.
+    """
+    return 0.0269 / (rn**0.14)
+
+
+def zero_lift_drag_coefficient(c_f: npt.NDArray[np.float_], psi_0: float) -> npt.NDArray[np.float_]:
+    """
+    Calculate zero-lift drag coefficient.
+
+    Parameters
+    ----------
+    c_f : npt.NDArray[np.float_]
+        Skin friction coefficient
+    psi_0 : float
+        Aircraft geometry drag parameter
+
+    Returns
+    -------
+    npt.NDArray[np.float_]
+        Zero-lift drag coefficient (c_d_0)
+
+    References
+    ----------
+    - Refer to Eq. (9) of Poll & Schumann (2021).
+    - Poll & Schumann (2021). An estimation method for the fuel burn and other performance characteristics of civil
+        transport aircraft in the cruise. Part 1: fundamental quantities and governing relations for a general
+        atmosphere. Aero. J., 125(1284), 296-340, doi: 10.1017/aer.2020.62.
+    """
+    return c_f * psi_0
+
+
+def oswald_efficiency_factor(
+        c_drag_0: npt.NDArray[np.float_], atyp_param: AircraftEngineParams
+) -> npt.NDArray[np.float_]:
+    """
+    Calculate Oswald efficiency factor.
+
+    The Oswald efficiency factor captures all the lift-dependent drag effects, including the vortex drag on the wing
+    (primary source), tailplane and fuselage, and is a function of aircraft geometry and the zero-lift drag coefficient.
+
+    Parameters
+    ----------
+    c_drag_0 : npt.NDArray[np.float_]
+        Zero-lift drag coefficient.
+    atyp_param : AircraftEngineParams
+        Extracted aircraft and engine parameters.
+
+    Returns
+    -------
+    npt.NDArray[np.float_]
+        Oswald efficiency factor (e_ls)
+
+    References
+    ----------
+    - Refer to Eq. (12) of Poll & Schumann (2021).
+    - Poll & Schumann (2021). An estimation method for the fuel burn and other performance characteristics of civil
+        transport aircraft in the cruise. Part 1: fundamental quantities and governing relations for a general
+        atmosphere. Aero. J., 125(1284), 296-340, doi: 10.1017/aer.2020.62.
+    """
+    numer = np.where(atyp_param.winglets, 1.075, 1)
```

Comment:
Is `atyp_params.winglets` just a python boolean here? If so, this can be

```python
numer = 1.075 if atyp_params.winglets else 1.0
```
or
```python
if atyp_params.winglets:
   numer = 1.075
else:
   numer = 1.0
```